src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime.Tests/PluginSourcesRepositoryTests.cs

```diff
@@ -0,0 +1,250 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Specialized;
+using Microsoft.Performance.Testing;
+using Microsoft.Performance.Toolkit.Plugins.Core.Discovery;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Tests
+{
+    [TestClass]
+    public class PluginSourcesRepositoryTests
+    {
+        [TestMethod]
+        [UnitTest]
+        public void Add_Single_Success()
+        {
+            var repo = new PluginSourceRepository();
+            var source = new PluginSource(FakeUris.Uri1);
+
+            bool success = repo.Add(source);
+
+            Assert.IsTrue(success);
+            Assert.AreEqual(1, repo.Items.Count());
+            Assert.AreSame(source, repo.Items.Single());
+        }
+
+        [TestMethod]
+        [UnitTest]
+        public void Add_Multiple_Success()
```

Comment:
I would make these `Success` tests parameterized tests with the `DataRow` attribute

Reply:
I don't think these are suitable for `DataRow` because these tests are actually each testing against only one input and verifying the behavior by a couple of asserts.

Reply:
```cs
public void Add_Success(Uri[] uris)
{
            var repo = new PluginSourceRepository();
            PluginSource[] sources = uris.Select(uri => new PluginSource(uri)).ToArray();
            IEnumerable<PluginSource> addedItems = repo. Add(sources );
            
            Assert.AreEqual(uris.Length, addedItems.Count());
            Assert.AreEqual(uris.Length, repo.Items.Count());

            foreach (PluginSource source in sources)
            {
                        Assert.IsTrue(addedItems.Contains(source);
                        Assert.IsTrue(repo.Items.Contains(source);
            }
}
```

Reply:
Then you would have two datarows. One would provide `Uri[] { FakeUris.Uri1 }`, the other all 3 URIs